### PR TITLE
Add support for MS Teams webhooks

### PR DESCRIPTION
**Pull request by dangrier**
Implements #5964 to add Microsoft Teams webhooks (Incoming Webhook Connector).

Largely matched styling of Discord webhooks and aligned with other webhook functionality/appearance.

Tested on a local instance, with a connection to Teams and functioned as expected.

All commits signed off, and latest changes integrated from upstream.

### CODE REVIEW

File: models/webhook_msteams.go [hunk 1]
```diff
@@ -0,0 +1,703 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
```
Comment (lunny) [inline models/webhook_msteams.go]: 2019

Comment (dangrier) [reply models/webhook_msteams.go]: Updated in 15b22b02da578d547b87c82b9139b4f8733ec76d thanks

**Comment by dangrier** [comment]
Is there anything else needed for approval?

**Comment by techknowlogick** [comment]
@dangrier could you resolve git conflicts, and post screenshots of what a webhook would look like in MS Teams?

**Comment by techknowlogick** [comment]
@dangrier I've just resolved the git conflicts. I did it via Github interface so hopefully a `make fmt` isn't needed.

**Comment by dangrier** [comment]
Okay, didn't see that so also resolved - but not to worry!

It looks like this:
image

**Review by techknowlogick — APPROVED** [review]

**Comment by dangrier** [comment]
The merge seems to have broken the webhook object, it isn't displaying as it was previously in Teams:

image

**Comment by dangrier** [comment]
Disregard, I realised that the webhook was set up when TELEGRAM wasn't before MSTEAMS in the `HookTaskType`s - so my local test Gitea DB is sending a Telegram webhook to Teams.

All is well :) thanks

**Comment by techknowlogick** [comment]
@dangrier ah phew. glad that was quick to diagnose.

**Review by lafriks — APPROVED** [review]

**Pull request merged**